lib/opus/pipeline/stage.ex

```diff
@@ -24,19 +24,22 @@ defmodule Opus.Pipeline.Stage do
       when is_atom(fun),
       do: maybe_run({module, type, name, %{opts | if: {module, fun, [input]}}}, input)
 
+  def maybe_run({module, :skip, name, %{if: {_m, _f, _a} = condition}}, input) do
+    case Safe.apply(condition) do
+      true ->
+        skip_stage({module, name}, input)
```

Comment:
This is tricky because in this case the whole pipeline is skipped, not just a single stage. I think we should use a different instrumentation event, something like `pipeline_skipped`.

Reply:
Ouch, really well noticed! Thanks again! Changed it, hope it's better now